promql/promqltest/README.md

```diff
@@ -73,8 +75,8 @@ When loading a batch of classic histogram float series, you can optionally appen
 
 ## `eval` command
 
-`eval` runs a query against the test environment and asserts that the result is as expected.
-It requires the query to succeed without any (info or warn) annotations.
+`eval` runs a query against the test environment and asserts that the result is as expected. 
+It requires the query to succeed without any failures unless an `expect fail` line is provided. Previously `eval` expected no `info` or `warn` annotation, but now `expect no_info` and `expect no_warn` lines must be explicitly provided.
```

Comment:
Not something for this PR, but it might be useful to introduce a "strict mode" for test files, where `expect no_info` and `expect no_warn` are implicitly enabled for all tests that don't explicitly expect annotations (ie. match the old behaviour).

Reply:
Good idea! We can handle that in a separate PR.